builder/dockerfile/dispatchers.go

```diff
@@ -152,9 +153,13 @@ func (d *dispatchRequest) getImageMount(imageRefOrID string) (*imageMount, error
 //
 func initializeStage(d dispatchRequest, cmd *instructions.Stage) error {
 	d.builder.imageProber.Reset()
-	if err := system.ValidatePlatform(&cmd.Platform); err != nil {
-		return err
+	if cmd.Platform.OS != "" {
+		_, err := platforms.Parse(cmd.Platform.OS)
+		if err != nil {
```

Comment:
nit; err could be scoped to the `if` here;

```go
if _, err := platforms.Parse(cmd.Platform.OS); err != nil {
    return err
}
```

Reply:
Nice catch 👍